apps/angular/signal-input/src/app/models/category.model.ts

```diff
@@ -0,0 +1,15 @@
+export enum CategoryType {
+  Youth = 'Youth',
+  Junior = 'Junior',
+  Open = 'Open',
+  Senior = 'Senior',
+}
+
+export class Category {
+  static ageToCategory(age: number): CategoryType {
+    if (age < 10) return CategoryType.Youth;
+    else if (age < 18) return CategoryType.Junior;
+    else if (age < 35) return CategoryType.Open;
+    return CategoryType.Senior;
+  }
+}
```

Comment:
From your code, we can see that you come from a backend OOP language like Java. Am I right? 😅

In TS, I always avoid enums. Enums were created for java developers but don't fit well in TS. Union string or map type is better. 

I don't see the value of the static function inside a class. A pure function inside a util file feels more natural and more functional. Less boilerplate

But both implementations work. Thanks for showing your implementation.

Reply:
No but i kind of like the Java syntax 😄 

I'll give it a try and refactor this, thanks for your advice!

Reply:
@tomalaforge Maybe thats an idea for a new typescript challenge like _Avoid Enums_?

Reply:
Yes we can create a challenge, that's right.
you can look at this video from TS guru: https://www.youtube.com/watch?v=jjMbPt_H3RQ&t=480s&ab_channel=MattPocock